### PR TITLE
Username Functionality added

### PR DESCRIPTION
**Pull request by Ansh101112**
## Fixes Issue
#1767

## Summary by CodeRabbit

- **New Features**
  - Added a `getProducts` function to fetch and return products as JSON.
  - Integrated search functionality accepting `searchTerm` and `handleSearch` as props.
  - Enhanced login response to include username.

- **Enhancements**
  - Updated product schema for better database interaction.
  - Improved login and logout UI, including displaying the username.
  - Refined button and navigation link styles for better user experience.

- **Bug Fixes**
  - Corrected local storage key from "isloggedin" to "isLoggedIn".

- **Chores**
  - Updated import statements for consistency.

### CODE REVIEW

File: backend/models/Product.js
Comment (codervivek5) [inline backend/models/Product.js:1]: Product description is missing and also there in no ratings also no discount?

File: src/components/SearchBar/SearchBar.jsx
Comment (codervivek5) [inline src/components/SearchBar/SearchBar.jsx:1]: Search functionality was properly working so you have also changed it ?

Comment (Ansh101112) [reply src/components/SearchBar/SearchBar.jsx:1]: No sir I just pulled the code didn't change anything in this there is a conflict which I fixed after pulling the code

File: backend/models/Product.js
Comment (Ansh101112) [reply backend/models/Product.js:1]: Okay will update the model in next pr as in this I have just added just a blueprint

**Comment by codervivek5** [comment]
mention the issue number and also close related to this PR

**Pull request merged**